milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
We could delete a bunch of code and save ourselves a DB query by removing the relationship type validation completely.

@mattdrayer Thoughts?